elastic/kibana
# [Security Solution] [Detection & Response] 131827 Update Detections Response view with pagination and opening numbers in timeline

#131827 

To test this PR locally: add `xpack.securitySolution.enableExperimental: [‘detectionResponseEnabled’]` to your `kibana.dev.yml`


@elastic/security-threat-hunting-investigations:
The change is exporting `getDataProvider`, to allow for navigating to the Timeline flyout.

Figma: https://www.figma.com/file/yGGkXDuUQzoR7faSVu6F9p/OOTB-Dashboard-for-analysts?node-id=1080%3A113129

![Recording 2022-05-17 at 16 31 29](https://user-images.githubusercontent.com/28942857/168913519-c929bac2-41e5-46b0-a878-a0f65dcfb5ae.gif)


Bugs fixed in this ticket:
- [x]  header font size of alerts chart needs to be uniform with other headers

## Enhancements:
- [x] Update headers of vulnerable tables to new name "hosts/users by alert severity" & add tooltip for max results
- [x] Add ability for clicking on numbers to link to a timeline
- [x] Change alert colour scheme to match current alert colours
- [x] Implement pagination for vulnerable host/user tables and remove buttons.

- [ ] Add cypress tests for testing navigation to timeline
- [ ] Remove feature flag?


## Background Conversations:
- [ ] Donut charts are not uniform in colour. A conversation has been initiated with the charts team and design to make further decisions on this: https://elastic.slack.com/archives/CFK9BJ248/p1651826940605369

## QA:
Information about the view:

For 8.3 We are only building the sections [noted in this document](https://docs.google.com/document/d/1qEo8A1WZbj6vmqIdwHAsMhixPHW1TIVevHXPg8AeAm0/edit) by the thumbs up (with green circle) further work is slated potentially for 8.4.

Please refer to that document for more information about this project, but find some things noted below:

- Open/Acknowledged/Closed hyperlinks in alerts chart are not clickable as the app does not have the ability to route to those specific sections in Alerts page. (we can for open, but not closed or acknowledged, so we have opted to make none of them clickable

-  Same as above with hyperlinks in cases chart 
- Alert / Alert counts for each table have been requested to link to timelines. For eg: 

<img width="572" alt="image" src="https://user-images.githubusercontent.com/28942857/167464374-bd7570c1-41dd-4a98-aadd-be952f8ec589.png">

clicking 7 would link to a timeline filtered by alerts by host.name: "Win 8"


Permissions
If a user does not have case permissions, no case chart or tables will be seen.
If a user does not have alerts permissions, no alert chart or tables will be seen.
If a user has not pertinent permissions, they will see a no permissioned view.


### Checklist
